server/src/main/java/org/opensearch/cluster/routing/allocation/AllocationService.java

```diff
@@ -617,10 +626,16 @@ private void allocateExistingUnassignedShards(RoutingAllocation allocation) {
 
     private void allocateAllUnassignedShards(RoutingAllocation allocation) {
         ExistingShardsAllocator allocator = existingShardsAllocators.get(ShardsBatchGatewayAllocator.ALLOCATOR_NAME);
-        allocator.allocateAllUnassignedShards(allocation, true);
+        BatchRunnableExecutor primaryShardsBatchExecutor = allocator.allocateAllUnassignedShards(allocation, true);
+        if (primaryShardsBatchExecutor != null) {
+            primaryShardsBatchExecutor.run();
```

Comment:
Wouldn't subsequent reroute be triggered if there's async fetch to process that were triggered by the executed batches? - https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/gateway/AsyncShardFetch.java#L220

Reply:
It will be triggered once asyncfetch for batch has response from datanode. But i don't think it is guaranteed to have pending fetches when the reroute times out. It might be well the case that all the responses have been already received and only processing is pending when reroute times out.

Reply:
Got it. Can we take this up in a follow up PR as this would need some extra work and would add extra complexity to current PR. Opened tracking issue - https://github.com/opensearch-project/OpenSearch/issues/14945